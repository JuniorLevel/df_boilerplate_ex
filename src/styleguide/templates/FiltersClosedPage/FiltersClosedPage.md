```jsx
import React, { useState } from 'react';
import {
	FiltersClosed,
	FiltersClosedListProvider,
	FiltersSearchListContainer,
	FiltersSearchListItem,
	Layout,
} from '@juniorlevel/df_boilerplate-library';

const buttons = [
	{
		id: 1,
		title: 'Orders',
		onClick: () => {},
	},
	{
		id: 2,
		title: 'Compare',
		onClick: () => {},
	},
	{
		id: 3,
		title: 'Admin',
		onClick: () => {},
	},
];

const data = [
	{
		id: 1,
		dataSearch: [],
		dataSearchUsers: [
			{
				id: 1,
				role: 'Creator',
				status: 'updated',
				username: 'Roman Smit',
			},
			{
				id: 2,
				role: 'Assigned',
				status: 'changed',
				username: 'John Doe',
			},
			{
				id: 3,
				role: 'Watcher',
				status: 'unmodified',
				username: 'Jerrard Doe',
			},
		],
	},
	{
		id: 2,
		dataSearch: [],
		dataSearchUsers: [
			{
				id: 1,
				role: 'Creator',
				status: 'updated',
				username: 'Roman Smit',
			},
			{
				id: 2,
				role: 'Assigned',
				status: 'changed',
				username: 'John Doe',
			},
			{
				id: 3,
				role: 'Watcher',
				status: 'unmodified',
				username: 'Jerrard Doe',
			},
		],
	},
	{
		id: 3,
		dataSearch: [],
		dataSearchUsers: [
			{
				id: 1,
				role: 'Creator',
				status: 'updated',
				username: 'Roman Smit',
			},
			{
				id: 2,
				role: 'Assigned',
				status: 'changed',
				username: 'John Doe',
			},
			{
				id: 3,
				role: 'Watcher',
				status: 'unmodified',
				username: 'Jerrard Doe',
			},
		],
	},
	{
		id: 4,
		dataSearch: [],
		dataSearchUsers: [
			{
				id: 1,
				role: 'Creator',
				status: 'updated',
				username: 'Roman Smit',
			},
			{
				id: 2,
				role: 'Assigned',
				status: 'changed',
				username: 'John Doe',
			},
			{
				id: 3,
				role: 'Watcher',
				status: 'unmodified',
				username: 'Jerrard Doe',
			},
		],
	},
	{
		id: 5,
		dataSearch: [],
		dataSearchUsers: [
			{
				id: 1,
				role: 'Creator',
				status: 'updated',
				username: 'Roman Smit',
			},
			{
				id: 2,
				role: 'Assigned',
				status: 'changed',
				username: 'John Doe',
			},
			{
				id: 3,
				role: 'Watcher',
				status: 'unmodified',
				username: 'Jerrard Doe',
			},
		],
	},
	{
		id: 6,
		dataSearch: [],
		dataSearchUsers: [
			{
				id: 1,
				role: 'Creator',
				status: 'updated',
				username: 'Roman Smit',
			},
			{
				id: 2,
				role: 'Assigned',
				status: 'changed',
				username: 'John Doe',
			},
			{
				id: 3,
				role: 'Watcher',
				status: 'unmodified',
				username: 'Jerrard Doe',
			},
		],
	},
];

const [filters, setFilters] = useState([
	{
		id: 1,
		title: 'Flow_typed-1',
		item: [{ id: 1, status: 'updated', text: 'lorem-1' }],
	},

	{
		id: 2,
		title: 'Flow_typed-2',
		item: [
			{ id: 1, status: 'changed', text: 'project-1' },
			{ id: 2, status: 'changed', text: 'project-2' },
			{ id: 3, status: 'updated', text: 'project-3' },
		],
	},
	{
		id: 3,
		title: 'Flow_typed-3',
		item: [
			{ id: 1, status: 'updated', text: 'issue-1' },
			{ id: 2, status: 'updated', text: 'issue-2' },
		],
	},
]);

<FiltersClosedListProvider>
	<Layout buttons={buttons} title="Caption" avatar={{ title: 'AV' }}>
		<FiltersClosed isOpen={false} filtersList={filters} />
		<FiltersSearchListContainer>
			<FiltersSearchListItem key={data[0].id} listData={data[0]} />
			<FiltersSearchListItem key={data[1].id} listData={data[1]} />
			<FiltersSearchListItem key={data[2].id} listData={data[2]} />
			<FiltersSearchListItem key={data[3].id} listData={data[3]} />
			<FiltersSearchListItem key={data[4].id} listData={data[4]} />
			<FiltersSearchListItem key={6} listData={null} />
		</FiltersSearchListContainer>
	</Layout>
</FiltersClosedListProvider>;
```

```jsx noeditor
import Image1 from '../../assets/FiltersClosed/fIlters-closed.png';
<img src={Image1} />;
```
