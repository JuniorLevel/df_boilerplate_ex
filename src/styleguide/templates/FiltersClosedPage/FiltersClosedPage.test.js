import React from 'react';
import { Context as ResponsiveContext } from 'react-responsive';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { FiltersClosedPage } from './FiltersClosedPage';
import '../../../../__mocks__/fileMock';

describe('FiltersClosedPage', () => {
	test('should render FiltersClosedPage component without crashing', () => {
		const { container } = render(<FiltersClosedPage />);
		expect(container).toBeInTheDocument();
	});
	test('matches the snapshot desktop', () => {
		const { container: desktop } = render(
			<ResponsiveContext.Provider value={{ width: 1920 }}>
				<BrowserRouter>
					<FiltersClosedPage />
				</BrowserRouter>
			</ResponsiveContext.Provider>
		);
		expect(desktop).toMatchSnapshot();
	});
	test('matches the snapshot laptop', () => {
		const { container: laptop } = render(
			<ResponsiveContext.Provider value={{ width: 1024 }}>
				<BrowserRouter>
					<FiltersClosedPage />
				</BrowserRouter>
			</ResponsiveContext.Provider>
		);
		expect(laptop).toMatchSnapshot();
	});
	test('matches the snapshot mobile', () => {
		const { container: mobile } = render(
			<ResponsiveContext.Provider value={{ width: 360 }}>
				<BrowserRouter>
					<FiltersClosedPage />
				</BrowserRouter>
			</ResponsiveContext.Provider>
		);
		expect(mobile).toMatchSnapshot();
	});
});
