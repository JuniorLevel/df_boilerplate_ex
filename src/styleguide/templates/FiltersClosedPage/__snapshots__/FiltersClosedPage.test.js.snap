// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersClosedPage matches the snapshot desktop 1`] = `
<div>
  <div
    class="ant-layout Layoutstyles__StyledLayout-sc-z125ah-0 ydqdt css-1c0na6j"
  >
    <div
      class="ant-layout css-1c0na6j"
    >
      <main
        class="ant-layout-content Layoutstyles__StyledContent-sc-z125ah-1 MFHYS css-1c0na6j"
      >
        <div
          class="FiltersSearchListContainer__StyledFiltersSearchContainer-sc-1du5nm9-0 QtmlH"
        />
      </main>
    </div>
  </div>
</div>
`;

exports[`FiltersClosedPage matches the snapshot laptop 1`] = `
<div>
  <div
    class="ant-layout Layoutstyles__StyledLayout-sc-z125ah-0 ydqdt css-1c0na6j"
  >
    <div
      class="ant-layout css-1c0na6j"
    >
      <main
        class="ant-layout-content Layoutstyles__StyledContent-sc-z125ah-1 MFHYS css-1c0na6j"
      >
        <div
          class="FiltersSearchListContainer__StyledFiltersSearchContainer-sc-1du5nm9-0 QtmlH"
        />
      </main>
    </div>
  </div>
</div>
`;

exports[`FiltersClosedPage matches the snapshot mobile 1`] = `
<div>
  <div
    class="ant-layout Layoutstyles__StyledLayout-sc-z125ah-0 ydqdt css-1c0na6j"
  >
    <div
      class="ant-layout css-1c0na6j"
    >
      <main
        class="ant-layout-content Layoutstyles__StyledContent-sc-z125ah-1 MFHYS css-1c0na6j"
      >
        <div
          class="FiltersSearchListContainer__StyledFiltersSearchContainer-sc-1du5nm9-0 QtmlH"
        />
      </main>
    </div>
  </div>
</div>
`;
